src/main/scala/firrtl/util/BackendCompilationUtilities.scala

```diff
@@ -104,12 +111,26 @@ trait BackendCompilationUtilities {
       }
     }
 
+    // Don't include the same file multiple times.
+    // If it's in BlackBoxSourceHelper.fileListName, don't explicitly include it on the command line.
+    // Build a map with absolute file paths as keys to use as a filter to exclude already included additional Verilog sources.
+    val blackBoxHelperFiles: Map[String, Boolean] = {
+      if(list_file.exists()) {
+        io.Source.fromFile(list_file).getLines.map {
+          line => line -> true
+        }.toMap
+      }
+      else {
+        Map.empty
+      }
+    }
+    val vSourcesFiltered = vSources.filterNot(f => blackBoxHelperFiles.contains(f.getAbsolutePath))
```

Comment:
The BlackBoxSourceHelper does write absolute paths so this will *probably* work in most cases, but I think it's brittle and possibly not portable to use Strings comparing absolute paths (eg. Windows). I think we should use proper java APIs to compare the files, eg.
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#isSameFile-java.nio.file.Path-java.nio.file.Path-

```
scala> import java.io.File
import java.io.File

scala> import java.nio.file.Files
import java.nio.file.Files

scala> val f = new File("build.sbt")
f: java.io.File = build.sbt

scala> val f2 = new File("project/../build.sbt")
f2: java.io.File = project/../build.sbt

scala> f == f2
res0: Boolean = false

scala> f.getAbsolutePath
res1: String = /Users/jack/console/build.sbt

scala> f2.getAbsolutePath
res2: String = /Users/jack/console/project/../build.sbt

scala> f.getCanonicalPath
res3: String = /Users/jack/console/build.sbt

scala> f2.getCanonicalPath
res4: String = /Users/jack/console/build.sbt

scala> Files.isSameFile(f.toPath, f2.toPath)
res6: Boolean = true
```

Reply:
Unfortunately, I'm not sure of a good way doing a constant time lookup for files that are the same using something like `Files.isSameFile`, perhaps there is a way